tests/user/test_user_auth exceptions.py

```diff
@@ -0,0 +1,124 @@
+    get_server_response_by_exception,
+    AuthorizationHeaderNotPresent,
+    AuthorizationHeaderEmptyContent,
+    InvalidAuthorizationHeader,
+    InvalidLambdaEventInput,
+    UserNotFound,
+    PasswordNotMatch,
+    InvalidUserPassword,
+    InvalidUserName,
+    InvalidDynamoResponse,
+    InconsistentExceptionDictionary,
+)
+
+_TEST_DUPLICATED_EXCEPTION_TO_HTTP_RESPONSE: Dict[Tuple, Dict] = {
+    (UserNotFound, PasswordNotMatch): {
+        "statusCode": 401,
+        "body": "User or password incorrect",
+    },
+    (
+        AuthorizationHeaderNotPresent,
+        UserNotFound,
+        InvalidAuthorizationHeader,
+    ): {
+        "statusCode": 400,
+        "body": "Invalid authentication input",
+    },
+}
+
+_TEST_EMPTY_EXCEPTION_LIST_RESPONSE: Dict = {
+    "statusCode": 500,
+    "body": "Internal Server Error",
+}
+
+
+@pytest.mark.parametrize(
+    "exception_type, expected_response",
+    [
+        (
+            UserNotFound,
+            {"statusCode": 401, "body": "User or password incorrect"},
+        ),
+        (
+            PasswordNotMatch,
+            {"statusCode": 401, "body": "User or password incorrect"},
+        ),
+        (
+            AuthorizationHeaderNotPresent,
+            {"statusCode": 400, "body": "Invalid authentication input"},
+        ),
+        (
+            AuthorizationHeaderEmptyContent,
+            {"statusCode": 400, "body": "Invalid authentication input"},
+        ),
+        (
+            InvalidAuthorizationHeader,
+            {"statusCode": 400, "body": "Invalid authentication input"},
+        ),
+        (
+            InvalidUserPassword,
+            {
+                "statusCode": 400,
+                "body": (
+                    "Username or password does not "
+                    + "match the minimal security requirements"
+                ),
+            },
+        ),
+        (
+            InvalidUserName,
+            {
+                "statusCode": 400,
+                "body": (
+                    "Username or password does not "
+                    + "match the minimal security requirements"
+                ),
+            },
+        ),
+        (
+            InvalidLambdaEventInput,
+            {"statusCode": 500, "body": "Internal Server Error"},
+        ),
+        (
+            InvalidDynamoResponse,
+            {"statusCode": 500, "body": "Internal Server Error"},
+        ),
+        (
+            KeyError,
+            {"statusCode": 500, "body": "Internal Server Error"},
+        ),
+        (
+            Exception,
+            {"statusCode": 500, "body": "Internal Server Error"},
+        ),
+    ],
+)
+def test_get_server_response_by_exception(
+    exception_type: Exception, expected_response: dict[str, Any]
+) -> None:
+    try:
+        raise (exception_type)
+    except Exception as exception:
+        assert expected_response == get_server_response_by_exception(exception)
```

Comment:
It looks like `exception_type` argument should be typed `Type[Exception]` currently.

You can rather modify your `@pytest.mark.parametrize` to instantiate objects of exceptions, for example `InvalidUserName` -> `InvalidUserName()` or instantiate in the body of the test by calling `exception_type()`, either works and would simplify your current instantiation using `raise`

the body would be 1 liner if you keep passing types to the test
```python
def test_get_server_response_by_exception(
    exception_type: Type[Exception], expected_response: dict[str, Any]
) -> None:
    assert expected_response == get_server_response_by_exception(exception_type())
```

Or also a 1 liner if you instantiate in pytest parameters
```python
def test_get_server_response_by_exception(
    exception: Exception, expected_response: dict[str, Any]
) -> None:
    assert expected_response == get_server_response_by_exception(exception)
```